apps/prairielearn/src/pages/userSettings/userSettings.html.ts

```diff
@@ -192,6 +222,64 @@ export function UserSettings({
   `.toString();
 }
 
+function PurchaseTable({ purchases }: { purchases: PurchaseRow[] }) {
+  return html`
+    <div class="table-responsive">
+      <table class="table">
+        <thead>
+          <tr>
+            <th>ID</th>
+            <th>Course</th>
+            <th>Date</th>
+            <th>Amount</th>
+            <th>Status</th>
+          </tr>
+        </thead>
+        <tbody>
+          ${purchases.map((purchase) => {
+            const courseName = purchase.course
+              ? `${purchase.course.short_name}: ${purchase.course.title}`
+              : 'Unknown course';
+
+            const courseInstanceName =
+              purchase.course_instance?.long_name ?? 'Unknown course instance';
+
+            return html`<tr>
+              <td>${purchase.stripe_checkout_session.id}</td>
+              <td>
+                <a
+                  ${purchase.course_instance == null
+                    ? ''
+                    : html`href="/pl/course_instance/${purchase.course_instance?.id}"`}
+                >
+                  ${courseName} (${courseInstanceName})
+                </a>
+              </td>
+              <td>${purchase.stripe_checkout_session.created_at?.toString()}</td>
+              <td>${formatStripePrice(purchase.stripe_checkout_session.data.amount_total)} USD</td>
+              <td>
+                ${StripeCheckoutSessionPaymentStatus({
+                  session: purchase.stripe_checkout_session,
+                })}
+              </td>
+            </tr>`;
+          })}
+        </tbody>
+      </table>
+    </div>
+  `;
+}
+
+function StripeCheckoutSessionPaymentStatus({ session }: { session: StripeCheckoutSession }) {
+  if (session.data.payment_status === 'paid') {
+    return html`<span class="badge badge-success">Payment received</span>`;
+  } else if (session.data.payment_status === 'unpaid') {
+    return html`<span class="badge badge-secondary">Pending</span>`;
+  } else {
+    return '';
```

Comment:
Should I return "unknown" here? Not sure if we want to leak bad state to users.

Reply:
I think that it's probably better to have an `unknown` or `indeterminate` state than to show nothing?

Reply:
Added in ed989b3.